Add/sitehealth enqueued assets

manuelRod opened the pull request:
This was the original plugin to be moved to modules:
https://github.com/audrasjb/site-health-audit-enqueued-assets/blob/master/site-health-audit-enqueued-assets.php

manuelRod committed:
Plugin https://github.com/audrasjb/site-health-audit-enqueued-assets/blob/master/site-health-audit-enqueued-assets.php imported as a module and some little refactoring.

manuelRod committed:
partial unit testing added.

manuelRod committed:
Adding more unit testing.

manuelRod committed:
adding missing tests.

pbearne commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,210 @@
+ */
+function aea_get_total_enqueued_scripts() {
+	$enqueued_scripts = false;
+	if ( get_transient( 'aea_enqueued_scripts' ) ) {
```
can we call get_transient once

pbearne commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,210 @@
+ */
+function aea_get_total_enqueued_styles() {
+	$enqueued_styles = false;
+	if ( get_transient( 'aea_enqueued_styles' ) ) {
```
can we call get_transient once

pbearne commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,210 @@
+				$enqueued_scripts[] = $src;
+			}
+		}
+		set_transient( 'aea_enqueued_scripts', $enqueued_scripts, 12 * HOUR_IN_SECONDS );
```
love to see a filter here for TTL maybe

And should we be invaliding the cache for theme change and plugin install

pbearne commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,210 @@
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued CSS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
```
in all the actions blocks
Can we add a button/link to reset the count
I can see that someone could be trying to improve the score but it not changing as the transient set to 1 day :-)

pbearne reviewed (changes requested):
The tests look really good well done
Most of the comments are around the invaliding the cache

manuelRod commented:
This is the WP issue: https://github.com/WordPress/performance/issues/25

manuelRod committed:
removing duplicated get_transient

manuelRod committed:
Adding filter for transient, and invalidate cache on switch_theme, activated_plugin, deactivated_plugin

manuelRod committed:
Adding user action to null transient cache.

manuelRod commented:
@pbearne ready for the second code review, I added everything we talked about.

pbearne commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,242 @@
+function clean_aea_audit_action() {
+	if ( isset( $_GET['action'] ) && 'clean_aea_audit' === $_GET['action'] ) {
+		check_admin_referer( 'clean_aea_audit' );
+		invalidate_cache_transients();
```
can/should we do a redirect hare to clear the querystring
wp_safe_redirect( remove_query_arg( 'action', 'clean_aea_audit' ) );

manuelRod replied:
you are right! fixed.

pbearne reviewed (changes requested):
I don't like leaving a query string with an action in the URL as it reruns the action on a page reload

ThierryA commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,242 @@
+ *
+ * @since 1.0.0
+ */
+function aea_audit_enqueued_scripts() {
```
All prefix needs to be updated to `perflab_`

ThierryA commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,242 @@
+		'test'        => 'enqueued_css_assets',
+	);
+
+	if ( $enqueued_styles > 10 ) {
```
This signal needs to be more robust. The number of stylesheets is somehow a week signal, what really matter is the totally size. We need to grab and sum the size of all stylesheets and define a threshold for the warning.

manuelRod replied:
@ThierryA I've opened the PR  in WP https://github.com/WordPress/performance/pull/54 we can follow up in there.

ThierryA replied:
Thank you @manuelRod, much appreciated. Left the same comments on the new PR to continue the conversation there.

ThierryA commented on modules/audit-enqueued-assets/load.php:
```diff
@@ -0,0 +1,242 @@
+		'test'        => 'enqueued_js_assets',
+	);
+
+	if ( $enqueued_scripts > 10 ) {
```
Same comment as for the stylesheet, we need to grab and sum the total size of assets loaded.

ThierryA reviewed (changes requested):
Thanks for working on this 😄   I believe this is still WIP although the this PR is not appearing as a "Draft". Nonetheless I left a few early comments without doing a detailed review yet. The approach to trigger an asset warning is quite weak at the moment and need some more work.

By the way, why opening a PR agains this fork rather than on the https://github.com/WordPress/performance repo, that makes it hard for others to follow along with the review. Could we continue the conversation on the main repo please?

manuelRod committed:
redirects after cleaning cache to site-health.php without query args

manuelRod committed:
All prefix needs to be updated to perflab_

manuelRod committed:
adding functionality to get resources sizes

manuelRod committed:
Adding filesize to the transient cache, and functionality related to it.

manuelRod committed:
Merge branch 'trunk' into add/sitehealth-enqueued-assets

manuelRod committed:
migrating module to new module structure

manuelRod committed:
migration to new module structure

manuelRod committed:
split code refactor, creation of a new helper file with helper methods and unit test reordering.

manuelRod committed:
prefix fix

manuelRod committed:
refactor adding suggestions:
- renaming transients
- view_site_health_checks capability check to avoid non logged and non view_site_health_checks users to trigger checks
- is_front_page check to avoid creating warnings in any random page.

manuelRod committed:
changing wording and adding information about bellow threshold limits.

manuelRod committed:
implementing new threshold limits

manuelRod committed:
Adding filters for limits. Fixing _n translatable strings.

manuelRod committed:
nit-pick fixes

manuelRod committed:
Adding new supported wp setups for perflab_aea_get_path_from_resource_url.
Changing hooks to capture enqueued scripts/styles and inlines.
